WBooks/src/app/screens/BookDetail/components/DetailBox/index.js

```diff
@@ -1,25 +1,39 @@
 import React, { Component } from 'react';
 import { withNavigation } from 'react-navigation';
+import { connect } from 'react-redux';
+import { actionCreators } from '@redux/wishlist/actions';
 
 import DetailBoxLayout from './layout';
 
 class DetailBox extends Component {
-  handleAddToWishlist = () => {};
+  book = null;
+
+  handleAddToWishlist = () => {
+    const { id } = this.book;
+    const { addToWishlist } = this.props;
+    addToWishlist(id);
+  };
+
+  handleRemoveBook = () => {
+    const { id } = this.book;
+    const { deleteFromWishlist } = this.props;
+    deleteFromWishlist(id);
+  };
 
   handleRentBook = () => {};
 
   render() {
-    const { navigation } = this.props;
-    const book = navigation.getParam('book', {});
+    const { navigation, wishlistBooks } = this.props;
+    this.book = navigation.getParam('book', {});
     const {
       image: { url: image },
       title,
       author,
       year,
-      genre
-    } = book;
-
-    console.log(image);
+      genre,
+      id
+    } = this.book;
+    const isOnWishlist = wishlistBooks.map(Id => Id.id).includes(id);
```

Comment:
We use words that start with capital letter only for Components, also i think instead of `Id` its `book`

Reply:
I would use find instead of map + includes